src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,708 @@
+        private static OpenApiMediaType CreateResponseContainer(string responseObjectSchemaName)
+        {
+            // schema for the response's collection of result records
+            OpenApiSchema resultCollectionSchema = new()
+            {
+                Reference = new OpenApiReference()
+                {
+                    Type = ReferenceType.Schema,
+                    Id = $"{responseObjectSchemaName}"
+                }
+            };
+
+            // Schema for the response's root property "value"
+            OpenApiSchema responseRootSchema = new()
+            {
+                Type = RESPONSE_ARRAY_PROPERTY,
+                Items = resultCollectionSchema
+            };
+
+            Dictionary<string, OpenApiSchema> responseBodyProperties = new()
+            {
+                {
+                    RESPONSE_VALUE_PROPERTY,
+                    responseRootSchema
+                }
+            };
+
+            OpenApiMediaType responsePayload = new()
+            {
+                Schema = new()
+                {
+                    Properties = responseBodyProperties
+                }
+            };
+
+            return responsePayload;
+        }
+
+        /// <summary>
+        /// Builds the schema objects for all entities present in the runtime configuration.
+        /// Two schemas per entity are created:
+        /// 1) {EntityName}      -> Primary keys present in schema, used for request bodies (excluding GET) and all response bodies.
+        /// 2) {EntityName}_NoPK -> No primary keys present in schema, used for POST requests where PK is autogenerated.
+        /// Schema objects can be referenced elsewhere in the OpenAPI document with the intent to reduce document verbosity.
+        /// </summary>
+        /// <returns>Collection of schemas for entities defined in the runtime configuration.</returns>
+        private Dictionary<string, OpenApiSchema> CreateComponentSchemas()
+        {
+            Dictionary<string, OpenApiSchema> schemas = new();
+
+            foreach (string entityName in _metadataProvider.EntityToDatabaseObject.Keys.ToList())
+            {
+                // Entities which disable their REST endpoint must not be included in
+                // the OpenAPI description document.
+                if (_runtimeConfig.Entities.TryGetValue(entityName, out Entity? entity) && entity is not null)
+                {
+                    if (entity.GetRestEnabledOrPathSettings() is bool restEnabled)
+                    {
+                        if (!restEnabled)
+                        {
+                            continue;
+                        }
+                    }
+                }
+
+                SourceDefinition sourceDefinition = _metadataProvider.GetSourceDefinition(entityName);
+                List<string> columns = /*sourceDefinition is null ? new List<string>() : */sourceDefinition.Columns.Keys.ToList();
```

Comment:
How about we use `HashSet` instead of `List` here? I see that we are performing remove operation in a loop at line 593, this would not be time efficient for a list I assume.